rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -326,11 +326,14 @@ def get_decoders_products(self) -> dict[str, list[EvmProduct]]:
 
         return possible_products
 
-    def reload_data(self, cursor: 'DBCursor') -> None:
+    def reload_data(self, cursor: 'DBCursor', reload_decoders: set[str] | None = None) -> None:
         """Reload all related settings from DB and data that any decoder may require from the chain
         so that decoding happens with latest data"""
         self.base.refresh_tracked_accounts(cursor)
-        for decoder in self.decoders.values():
+        for decoder_name, decoder in self.decoders.items():
+            if reload_decoders and decoder_name not in reload_decoders:
+                continue
```

Comment:
maybe something smarter to avoid iterating over all of them would be better like iterate over the set and do a get in self.decoders and then reload those only. Otherwise you have a lot of potential decoders and you basically continue in the loop

Reply:
yeah but this way I can reuse the same function and this happens very rarely. But yes this is an improvement to do